pytorch_lightning/accelerators/accelerator.py

```diff
@@ -114,7 +114,7 @@ def pre_dispatch(self, trainer: 'pl.Trainer') -> None:
     def _move_optimizer_state(self) -> None:
         """ Moves the state of the optimizers to the GPU if needed. """
         for opt in self.optimizers:
-            state = defaultdict(dict)
+            state: DefaultDict = defaultdict(dict)
```

Comment:
@awaelchli @Borda @carmocca do you know why this wasn't failing in https://github.com/PyTorchLightning/pytorch-lightning/pull/7277 ?

Reply:
because mypy check is not enforced and auto merge just merged it.

Reply:
Should it be enforced?

Reply:
I think it should now that it's green.

cc: @Borda